SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1194,6 +1194,20 @@ void USpatialActorChannel::UpdateEntityACLToNewOwner()
 	}
 }
 
+bool USpatialActorChannel::shouldChangeAuthorityIntent()
+{
+	// The authority intent should be changed by this worker if the following conditions are satisfied:
+	//  - This worker has authority over the authority intent component
+	//  - The authority intent should be changed, this could be because either:
+	//    - The actor no longer should be delegated to the virtual worker it currently is according to the load balancing strategy
+	//    - The actor is not delegated to a virtual worker
+	//  - The actor is not locked
+	return NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::AUTHORITY_INTENT_COMPONENT_ID) &&
+		(NetDriver->LoadBalanceStrategy->ShouldRelinquishAuthority(*Actor) ||
```

Comment:
Nit: Perhaps pulling this or out into another names bool variable might make the code easier to read?

bool bAuthorityIntentNeedsUpdating = NetDriver->LoadBalanceStrategy->ShouldRelinquishAuthority(*Actor) || etc 

Depends on your preference.

Reply:
I am not really sure here but if LockingPolicy->IsLocked is "simpler" than LoadBalanceStrategy->ShouldRelinquishAuthority it should be moved before ShouldRelinquishAuthority in the && condition.

Reply:
pulled out to local variable, and reordered just in case (Although I guess doing both of these might be putting a lot of faith in the compiler?)

Reply:
IsLocked was intentionally afterwards since the ShouldRelinquishAuthority will typically be false (and exit early) whereas !IsLocked will nearly always be true and go onto the ShouldRelinquishAuthority anyway

Reply:
reordered again and rephrased this as a comment, thanks

Reply:
Hmm how often do we expect HasAuthority to return false Ally?